test/for.civet

```diff
@@ -774,10 +774,23 @@ describe "for", ->
         . "NaN"
       x+1
     ---
-    let results=[];for (const x of y) {
+    let results;results=[];for (const x of y) {
       if (!(x != null)) { results.push(null);continue }
       if(isNaN(x)) { results = [
           "NaN"];break}
       results.push(x+1)
     };const values =results
   """
+
+  testCase """
+    break/continue with outside expression context
+    ---
+    for x of y
```

Comment:
Maybe this should warn?

Reply:
Hmm, good question. It does feel a little worrying throwing away code (the `with` arguments), though I think it's nicer than failing, in particular as you're transforming code to/from expression mode.

We don't currently have a warning system. @STRd6 Should we implement one? Perhaps `type: "Warning"` nodes, or `warning: true`/`subtype: "Warning"` on `type: "Error"` nodes? This would require some restructuring of current error handling, to report when possible but fail only in certain cases.

Reply:
Seems like something we'd like eventually. We can use a subtype on Error nodes for now.

Reply:
Sounds good. If it's OK with you, I'd like to leave this for a future project. Actually checking for `break with`/`continue with` that isn't consumed in a loop is trickier.

I also realize that `break :label with foo` doesn't currently work. (Currently you can only use `break with`/`continue with` to modify the parent loop.) Maybe also good for a future PR.